File: bash/code_to_gast/bash_main.py
import bashlex
import bash.code_to_gast.bash_router as bash_router

def bash_to_gast(bash_input):
    input_ast = ''
    try:
        input_ast = bashlex.parse(bash_input)
        print(input_ast)
    except:
        return 'Error: code could not compile'
    
    return bash_router.node_to_gast(input_ast)
    

File: bash/code_to_gast/bash_main.js
const bash_router = require("./bash_router.js")
const parse = require("bash-parser");

const bash_input = process.argv[2]
let input_ast = ""
try {
    input_ast = parse(bash_input);
} catch(err) {
    return "Error: code could not compile"
}

const gast = bash_router.node_to_gast(input_ast)
const json_string = JSON.stringify(gast)
console.log(json_string)
